# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmJS.cpp
# Commit: 662cb63c42d7
# Full Hash: 662cb63c42d7874a79256a6edf1a03d83e43a014
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2023-08-05 09:19:01
# Regressor Bug: 1692065
# File Overlap Count: 2
# Description:
#   Bug 1692065 - wasm: Create a unique wasm::AnyRef type. r=yury,jonco
#   
#   Right now wasm references area represented as a JSObject*. This is assumed in
#   many places in SpiderMonkey, most prominently in our GC integration
#   around tracing, the store buffer, and write barriers. This will cause issues
# ==============================================================================

diff -r 8b506ab41451 -r 662cb63c42d7 js/src/wasm/WasmJS.cpp
--- a/js/src/wasm/WasmJS.cpp	Fri Aug 04 13:54:57 2023 +0000
+++ b/js/src/wasm/WasmJS.cpp	Fri Aug 04 14:06:49 2023 +0000
@@ -3611,9 +3611,9 @@
   for (size_t i = 0; i < params.length(); i++) {
     ValType paramType = params[i];
     if (paramType.isRefRepr()) {
-      GCPtr<JSObject*>* objectPtr =
-          reinterpret_cast<GCPtr<JSObject*>*>(typedMem + offsets[i]);
-      TraceNullableEdge(trc, objectPtr, "reference-obj");
+      GCPtr<wasm::AnyRef>* paramPtr =
+          reinterpret_cast<GCPtr<AnyRef>*>(typedMem + offsets[i]);
+      TraceNullableEdge(trc, paramPtr, "wasm exception param");
     }
   }
 }